pennylane/shadows/classical_shadow.py

```diff
@@ -316,22 +317,139 @@ def qnode(x):
 
         return qml.math.squeeze(results)
 
+    def entropy(self, wires=[0], snapshots=None, alpha=2, k=1, base=None, atol=1e-5):
```

Comment:
Ah that was the problem. Is there a good intuition why providing `[0]` as a default is dangerous? Asking out of curiosity :)

Reply:
It's because `[0]` is mutable. If somewhere in the implementation you did something like:
```python
def entropy(self, wires=[0], ...):
    ...
    wires.append(1)
    ...
```
then the next call to `entropy()` would have a default value of `[0, 1]` instead of `[0]`. The replacement with `None` is a Python convention to prevent any possible bugs in the future.

Reply:
ah nice, thanks for the great example!